boomi/boomi-workflow.xml

```diff
@@ -0,0 +1,152 @@
+                    <CallEndpoint url="https://api.boomi.com/api/rest/v1/${/account_id}/AuditLog/queryMore" method="POST" savePath="/response" >
+                        <!-- <SSLConfiguration allowUntrustedServerCertificate="true" /> -->
+                        <BasicAuthentication username="${/username}" password="${/token}" />
+                        <RequestBody type="text/xml" encoding="UTF-8">${/query_token}</RequestBody>
+                    </CallEndpoint>
+
+                    <Log type="INFO" message="[Boomi]: Done fetch MORE AuditLogs, got status code: ${/response/status_code}." />
+
+                    <If condition="/response/status_code = 200">
+                        <XPathQuery xmlPath="/response/body" xPathQuery="/*['QueryResult'=local-name()]/*['result'=local-name()]" savePath="/events" />
+                        <XPathQuery xmlPath="/response/body" xPathQuery="/*['QueryResult'=local-name()]/*['result'=local-name()]/@date" savePath="/events_date" />
+                        <XPathQuery xmlPath="/response/body" xPathQuery="/*['QueryResult'=local-name()]/*['queryToken'=local-name()]/text()" singleton="true" savePath="/query_token" />
+
+                        <Log type="INFO" message="[Boomi]: Recived token for more events: ${not empty(/query_token)}" />
+
+                        <Set path="/events_count" value="${count(/events)}" />
+                        <Set path="/total_fetched_events" value="${/total_fetched_events + /events_count}" />
+
+                        <Log type="INFO" message="[Boomi]: Total exist events: ${/total_exist_events}." />
+                        <Log type="INFO" message="[Boomi]: Run number ${/fetch_counter} - fetched events: ${/events_count}." />
+                        <Log type="INFO" message="[Boomi]: Total fetched events: ${/total_fetched_events}." />
+                        <Log type="INFO" message="[Boomi]: More events to fetch: ${/total_exist_events - /total_fetched_events}." />
+
+                        <Set path="/fetch_counter" value="${/fetch_counter + 1}" />
+
+                        <Log type="INFO" message="[Boomi]: Posting fetched events..." />
+
+                        <PostEvents path="/events" source="${/identifier}" />
+                    </If>
+
+                    <Else>
+                        <Log type="ERROR" message="[Boomi]: status code ${/response/status_code}, abort to get events. Reason: ${/response/body}" />
+                        <Abort reason="${/response}" />
+                    </Else>
+
+                </While>
+                <Log type="INFO" message="[Boomi]: Done fetch period, total time in milliseconds is: ${time() - /current_time}" />
+
+                <If condition="${/total_exist_events} = ${/total_fetched_events}">
+                    <If condition="max(/events_date) > /bookmark">
+                        <Set path="/bookmark" value="${max(/events_date)}" />
```

Comment:
this bookmark should contain the max time for all loops in the pagination. correct?